dev-docs/source/WritingPerformanceTests.rst

```diff
@@ -0,0 +1,154 @@
+.. _WritingPerformanceTests:
+
+=========================
+Writing Performance Tests
+=========================
+
+.. contents::
+  :local:
+
+Overview
+########
+
+The point of Performance Tests is to track the performance of a piece of
+code through time, so that it will be easy to pinpoint any change that
+reduced the performance of a particular bit of code. Performance Tests
+will be built and run by a build job, that will track the history of
+timing vs revision. The Jenkins job that runs the performance tests
+sends out an email when the performance of a test is lowered by more
+than a threshold percentage (currently 25%), relative to the average of
+a few previous tests.
+
+How to Write C++ Performance Tests
+##################################
+
+An ideal performance test is neither too fast, nor too slow. The
+precision of timing will be insufficient if the test runs in much less
+than a second; tests that run for several minutes should also be
+avoided.
+
+C++ performance tests are written in the same way as unit tests, and in
+the same file as the unit tests for a particular class, except that you
+will add *Performance* to the end of the name of the test suite. For
+example, in MyAlgorithmTest.h:
+
+.. code-block:: c++
+
+   class MyAlgorithmTest : public CxxTest::TestSuite {
+      // Put in your usual, quick unit tests here
+   };
+   
+   class MyAlgorithmTestPerformance : public CxxTest::TestSuite {
+   public:
+      MatrixWorkspace_sptr WS;
+      int numpixels;
+       
+      void setUp() {
+         // Put in any code needed to set up your test,
+         // but that should NOT be counted in the execution time.
+      }
+       
+      void tearDown() {
+         // Clean-up code, also NOT counted in execution time.
+      }
+       
+      void test_slow_performance() {
+         // Put in a unit test that will be slow.
+      }
+   };
+
+Only the presence/absence of the word Performance is used to determine
+if it is a Performance test. As with unit tests, your suite name has to
+match the file name.
+
+You may include ASSERT's in the same way as a unit test to check that
+results are meaningful, but that is unnecessary since these tests are
+for performance, not function. Avoid adding so many assert's that it
+would slow down your test.
+
+Running Performance Tests
+#########################
+
+Performance tests targets are not added by default when running
+cmake/make. To add them as ctest targets, enable them with the flag:
+
+.. code-block:: sh
+
+cmake -DCXXTEST_ADD_PERFORMANCE=TRUE
```

Comment:
Indentation is needed on the examples here, currently they render as empty code blocks.